opm/simulators/linalg/PreconditionerFactory_impl.hpp

```diff
@@ -186,12 +194,23 @@ struct StandardPreconditioners
         // with the AMG hierarchy construction.
         if constexpr (std::is_same_v<O, Dune::OverlappingSchwarzOperator<M, V, V, C>>) {
           F::addCreator("amg", [](const O& op, const P& prm, const std::function<V()>&, std::size_t, const C& comm) {
+            using PrecPtr = std::shared_ptr<Dune::PreconditionerWithUpdate<V, V>>;
             const std::string smoother = prm.get<std::string>("smoother", "ParOverILU0");
             if (smoother == "ILU0" || smoother == "ParOverILU0") {
               using Smoother = Opm::ParallelOverlappingILU0<M, V, V, C>;
               auto crit = AMGHelper<O,C,M,V>::criterion(prm);
               auto sargs = AMGSmootherArgsHelper<Smoother>::args(prm);
-              return std::make_shared<Dune::Amg::AMGCPR<O, V, Smoother, C>>(op, crit, sargs, comm);
+              PrecPtr prec = std::make_shared<Dune::Amg::AMGCPR<O, V, Smoother, C>>(op, crit, sargs, comm);
+              return prec;
+            }
+            else if (smoother == "DILU") {
+              using Smoother = Dune::SeqDilu<M, V, V>;
+              using ParSmoother = Dune::BlockPreconditioner<V, V, C, Smoother>;
```

Comment:
Suggest changing these two typenames: `Smoother -> SeqSmoother` and `ParSmoother` -> `Smoother` (also remember to change on lines 209 and 212). Then the Smoother is the one actually used, the SeqSmoother is just a building block and not the other way around.

Reply:
Good suggestion, changed it now.